acceptancetests/jujupy/client.py

```diff
@@ -2121,6 +2128,90 @@ def switch(self, model=None, controller=None):
             raise ValueError('No target to switch to has been given.')
         self.juju('switch', (':'.join(args),), include_e=False)
 
+    def scp(self, _from, _to):
+        self.juju('scp', (_from, _to))
+
+
+# caas `add-k8s` did not implement parsing kube config path via flag,
+# so parse it via env var ->  https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L44
+KUBE_CONFIG_PATH_ENV_VAR = 'KUBECONFIG'
+
+
+class CaasClient(ModelClient):
```

Comment:
I'm not convinced this is the right way to achieve this split.

It adds complexity with the Client type selection, and it's possible there are edge cases of ```ModelClient``` creation that aren't covered here.

Perhaps we use composition instead. Something like this:
```python
class CaasClient:
    cloud_name = '...''
    ....

    def __init__(self, client):
        """:param client: ModelClient that is wrapping us."""
        self._client = client

    def add_cloud(self):
        self._client.juju('add-k8s', self.cloud_name)


# The existing ModelClient.
class ModelClient:

    def __init__(self, ...):
        self.caas = CaasClient(self)


client = ModelClient(...)
client.caas.add_cloud()
```

Thoughts?

Reply:
I think `CaasClient` and `ModelClient` are two similar but different, or you can think `Caas` model is a `special` model. So ModelClient should not have anything related Caas, but Caas should have all the attr/abilities of ModelClient and has some other stuff that ModelClient does `not` have.